src/test/java/uk/gov/pay/api/utils/mocks/ConnectorDDMockClient.java

```diff
@@ -72,7 +73,25 @@ public void respondOk_whenCreateAgreementRequest(String mandateId, MandateType m
                         ))
                 );
     }
-
+    public void respondOk_whenGetAgreementRequest(String mandateId, MandateType mandateType, String returnUrl, MandateState state, String gatewayAccountId, String chargeTokenId) {
+        whenGetAgreement(mandateId, gatewayAccountId)
+                .respond(response()
+                        .withStatusCode(200)
+                        .withHeader(CONTENT_TYPE, APPLICATION_JSON)
+                        .withBody(buildGetAgreementResponse(
+                                mandateId,
+                                mandateType,
+                                returnUrl,
+                                state,
+                                validGetLink(mandateLocation(gatewayAccountId, mandateId), "self"),
+                                validGetLink(nextUrl(chargeTokenId), "next_url"),
+                                validPostLink(nextUrlPost(), "next_url_post", "application/x-www-form-urlencoded",
+                                        new HashMap<String, String>() {{
+                                            put("chargeTokenId", chargeTokenId);
+                                        }})
```

Comment:
We could probably use `Collections.singletonMpa("chargeTokenId", chargeTokenId)` here but this is consistent with the rest of the test